metricbeat/module/windows/perfmon/reader.go

```diff
@@ -94,6 +93,32 @@ func NewReader(config Config) (*Reader, error) {
 	return r, nil
 }
 
+// RefreshCounterPaths will recheck for any new instances and add them to the counter list
+func (r *Reader) RefreshCounterPaths() error {
+	for _, counter := range r.config.CounterConfig {
+		if !strings.Contains(counter.Query, "*") {
+			continue
+		}
+		childQueries, err := r.query.ExpandWildCardPath(counter.Query)
+		if err == nil && len(childQueries) >= 1 && !strings.Contains(childQueries[0], "*") {
+			for _, v := range childQueries {
+				if err := r.query.AddCounter(v, counter, len(childQueries) > 1); err != nil {
+					r.query.Close()
```

Comment:
`RefreshCounterPaths()` is run on every fetch after the first one. If `AddCounter` fails, `r.query` will be closed and never reopened. Will then all the subsequent fetches fail because the query is closed?
I think that we shouldn't close here, or if we do it, we should have some reopening logic during fetch.

Reply:
Good catch @jsoriano , I used to call OpenQuery at each refresh but moved it out, I have removed the close option.